onnxruntime/core/providers/xnnpack/nn/max_pool.cc

```diff
@@ -193,9 +197,19 @@ MaxPool::MaxPool(const OpKernelInfo& info)
                                               stride_height, stride_width,
                                               dilation_height, dilation_width,
                                               output_min, output_max, flags, &p);
+  } else if (input_dtype == ONNX_NAMESPACE::TensorProto_DataType_FLOAT16) {
+    maxpool_type_ = OpComputeType::op_compute_type_fp16;
+    const float output_min = -65504.0;
+    const float output_max = 65504.0;
+    status = xnn_create_max_pooling2d_nhwc_f16(input_padding_top, input_padding_right,
+                                               input_padding_bottom, input_padding_left,
+                                               pooling_height, pooling_width,
+                                               stride_height, stride_width,
+                                               dilation_height, dilation_width,
+                                               output_min, output_max, flags, &p);
   } else {
     auto stype = DataTypeImpl::ToString(DataTypeImpl::TypeFromProto(*X_arg.TypeAsProto()));
-    ORT_THROW("unsupported Conv in maxpool, we have FLOAT|UINT8, but got ", stype);
+    ORT_THROW("unsupported Conv in maxpool, we have FLOAT|UINT8|FP16, but got ", stype);
```

Comment:
```suggestion
    ORT_THROW("unsupported Conv in maxpool, we have FLOAT|UINT8|FLOAT16, but got ", stype);
```

The values here match the last segment of the enum names, so it's easier to make the connection to the enum if we follow that pattern.